torch/csrc/jit/runtime/instruction.h

```diff
@@ -35,6 +35,7 @@ namespace jit {
   _(WAIT, "") /* wait for a future to be complete */                        \
   _(CALL, "F") /* call function X */                                        \
   _(GUARD, "T") /* check a guard against type_table, true if passes */      \
+  _(TYPECHECK, "T") /* check a guard against type_table, true if passes */  \
```

Comment:
I'm not sure "T" is the correct signature here. Could you please elaborate why are we using it? Should it be "TN" since we're inserting `num_inputs` as well? In either case, could you please expand the comment to cover the semantics of the op in more details please?